### ISSUE TITLE
Icons are mostly empty when not selected in the menubar - dark UI mode or resume related

### ISSUE THREAD

**Issue by ssbarnea** [ptsochantaris/trailer#260]
That's happening on OS X 10.11 and it seems to be like this for quite a few weeks. Restarts and even a recent Trailer.app update didn't fix this. 

If I go with the mouse over it will draw them properly, but when not selected they are almost fully transparent, only the red text being visible.

<img width="130" alt="screenshot 2016-05-27 09 28 29" src="https://cloud.githubusercontent.com/assets/102495/15602538/b5029ac8-23ed-11e6-96af-a2dee5abcfa6.png">
<img width="85" alt="screenshot 2016-05-27 09 28 08" src="https://cloud.githubusercontent.com/assets/102495/15602539/b5ef0566-23ed-11e6-8548-7c817db21078.png">

**Comment by ptsochantaris** [comment]
Hi and thanks for reporting this!

Just to clarify: When you say they are always invisible until you move the mouse over, do you mean just hovering the mouse, or selecting the menu-bar items like in the screenshot?

Also, I notice that Trailer seems to be running in dark mode but your menu-bar isn't in dark mode, which is weird? When Trailer detects dark mode on a system, it will draw its menu icons as white to contrast with the dark menu bar. This is what's causing your issue, but I'm wondering what's causing this mis-detection.

So first off, to help clarify, is your system set to run in dark mode in the first place? If not, it's a question of debugging Trailer's mode-detection of your setting. You *are* running in dark mode, then we need to figure out why your menu-bar is white :)

(PS Trying to reproduce your issue here I managed to find that the latest test build "lags" behind the UI dark mode for some reason, so your report is already leading to at least one fix :))

**Comment by ssbarnea** [comment]
I can assure you that this happens while I am not running in the dark-mode, only Trail.app believes that is in the dark mode. I need to click on it in order make it visible. 

If I remember well my laptop is configured to auto-activate night mode based on the hour and I am also using Flux.app and Bartender.app which could interact which the behaviour of the tray. 

My impression is that this is not related to the other apps but that is related to the dark mode detection. It seems that Trail.app is missing to detect switches between dark / day mode, probably in context to sleep/resume. I do not restart or even re-login on the laptop for weeks, usually I restart it only when a new OS X updated appears.

I manually quit Trail.app and restarted it and it looks ok. This indicates that is probably only the failure to switch modes.

**Comment by ptsochantaris** [comment]
Awesome, thank you for that detailed explanation, it makes a lot of things much clearer and I think your issue related to the breakage I just spotted.

If I'm right, then you can recreate the issue on your current build by just switching light/dark mode manually from your settings, and as you suspected, this is 100% Trailer's fault (and a liiiiittle bit of Appkit's for being weird when reporting the current UI appearance before an app has displayed anything :))

Could you have a go with the latest RC? It may fix your issue. (But please, when testing anything pre-release, export your Trailer settings first, and back-up your ~/Library/Application Support/com.housetrip.Trailer directory so you can revert to a previous version, *just in case* bad things happen)

https://www.dropbox.com/s/vjgugvohdy156t5/trailer-1323-rc2.zip?dl=0

**Comment by ssbarnea** [comment]
I already installed it and I will be back with the results, it may take a couple of days in order to reproduce the same conditions.

**Comment by ptsochantaris** [comment]
Thanks! Looking forward to your feedback.

**Comment by ssbarnea** [comment]
Still reproduces with build 1313. Unrelated to that, my impression is that the number of pull requests became bold and the number of notifications remained with regular font.

**Comment by ptsochantaris** [comment]
Hi! Ok, looks like we need to do some more digging, but we'll get there.

First off, just wanted to get a better idea of the PR count font change you mentioned. I can't seem to see that, but I wonder if I misunderstood which count you're referring to?

<img width="455" alt="screen shot 2016-05-29 at 11 32 45" src="https://cloud.githubusercontent.com/assets/521581/15632653/42462270-2591-11e6-905f-7d611b29bd31.png">

If you mean the status bar, the PR/Issue count will be bold only when it is red as well - if you're seeing something different then something may be wrong (or: you're not using any UI theming app by any chance?)

Now onto the main subject, I've got a feeling that whatever utility you're using to switch modes isn't sending the proper notification (or I should say, the proper notification isn't arriving for whatever reason) that the UI mode has changed. Can you try a couple of things for me?

- Sanity check, just in case: Manually switch light/dark mode on your system. Does Trailer then respect the mode (without restarting it?)
- Can you let me know the utility you are using for the automatic light/dark switching? May help to reproduce this locally.
- Can you remember if this occurs after your system wakes up? It may be that Trailer needs to "double-check" the dark mode setting after the system wakes up, as the other utility may switch modes in an OS mode where the notification gets lost.

Here's a link to the latest build, just so we can be in sync: https://www.dropbox.com/s/l0b8ug5zgcxb68o/trailer-140-rc3.zip?dl=0

**Comment by ptsochantaris** [comment]
Just keeping the RC build fresh: https://www.dropbox.com/s/51qmlo3hzmgo49w/trailer-140-rc5.zip?dl=0

**Comment by ssbarnea** [comment]
Just upgraded to rc5. Here are some additional feedback for the previous version: my impression is that happens only after a over-the-night sleep-wake cycle. The night mode is not activated on the computer at all, but once I login back in the morning the app believes that is in night mode. 

I am wondering if is not easier to check of the dark mode every 5 minutes and to make a decision based on that.

**Comment by ptsochantaris** [comment]
Hi, and thanks for trying it out.

From what you're describing, it sounds as if the app is receiving a dark mode notification while in power-nap but somehow never gets the notification that dark mode is now off in the morning.

Checking for dark mode every once in a while (e.g. during a refresh) could help fix this, but it's super-clunky and I'd like to avoid it if possible, if anything because you'd still see broken icons after your system wakes up.

But, yes, in essence we want to basically work around a lost UI notification. It sounds to me more like Trailer needs to check the state of the display few moments after the system has woken up from sleep, to determine if the UI appearance state has changed. I've seen similar issues in the past with WiFi availability being "strange" after a system sleep, so Trailer has to manually check it after a wake event.

Let's have another go. This build double-checks the UI state a second after the system announces a wake event, and will also ignore any UI changes during a power-nap. We may have to extend this perhaps, but worth a try in the meantime: https://www.dropbox.com/s/5f8ubdm99xsianu/trailer-140-rc6.zip?dl=0

**Comment by ptsochantaris** [comment]
Keeping the build fresh with some last-minute tweaks: https://www.dropbox.com/s/0vy182l44qy4qyi/trailer-140-rc7.zip?dl=0

**Comment by ssbarnea** [comment]
Update from this morning resume: for the first time, I see the icons correctly. This could be an indication that rc7 solved the problem.

**Comment by ptsochantaris** [comment]
It's certainly encouraging! Let's hope it keeps this up. Thanks for trying this out and keeping me up to date. Let's see if this behaviour stays consistent across the next few days.

Meanwhile, just to be sure you're using as stable a build as possible, here's the latest RC with the latest cleanups (I don't foresee too many more TBH, and there are no further tweaks to the code relating to your issue) https://www.dropbox.com/s/6vo013rt86h3imt/trailer-140-rc8.zip?dl=0

**Comment by ptsochantaris** [comment]
Just keeping the RC fresh: https://www.dropbox.com/s/fvilf3yeovx1xni/trailer-140-rc9.zip?dl=0

**Comment by ptsochantaris** [comment]
Hoping that the latest builds have "stayed" good for you! If all seems well, feel free to update to the final 1.4.0 which is up now, and you can find it from the releases page or using the "check for updates now" button in Trailer prefs. I'll be keeping this issue open until we're sure it is corrected, though.

**Comment by ssbarnea** [comment]
Thanks! Since the rc6 or rc7 I wasn't able to replicate the problem so I would mark the bug as solved!

**Comment by ptsochantaris** [comment]
That's great to know! Feel free to open a new issue in case you encounter this again or any other issue.
